src/arrayMethodJoin.js

```diff
@@ -4,9 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (!this.length) {
+      return '';
+    }
+
+    let result = String(this[0]);
+
+    if (this[0] === null || this[0] === undefined) {
+      result = '';
+    }
```

Comment:
```suggestion
    let result = '';
```
Ця перевірка на перший елемент не потрібна. Ти можеш починати цикл з нульового індексу (`for (let i = 0; i < this.length; i++)`) і тоді ти пройдешся і по першому елементу також. Ця логіка в тебе присутня і всередині циклу (21-й рядок)

Reply:
Можна перевіряти, якщо елемент не останній, то не додавати сепаратор. І робити це всередині циклу

Reply:
тодi так